tests/data/metrics/confusion_matrix/test_confusion_matrix_data_row.py

```diff
@@ -0,0 +1,57 @@
+from pytest_cases import fixture_ref
+from pytest_cases import pytest_parametrize_plus, fixture_ref
+
+from labelbox.data.metrics.confusion_matrix.confusion_matrix import confusion_matrix_metric
+
+
+@pytest_parametrize_plus("tool_examples", [
+    fixture_ref('polygon_pairs'),
+    fixture_ref('rectangle_pairs'),
+    fixture_ref('mask_pairs'),
+    fixture_ref('line_pairs'),
+    fixture_ref('point_pairs')
+])
+def test_overlapping_objects(tool_examples):
+    for example in tool_examples:
+        score = confusion_matrix_metric(example.ground_truths,
+                                        example.predictions)
+
+        if len(example.expected) == 0:
+            assert len(score) == 0
+        else:
+            expected = [0, 0, 0, 0]
+            for expected_values in example.expected.values():
+                for idx in range(4):
+                    expected[idx] += expected_values[idx]
```

Comment:
isnt this whole block just re-generating a list, `expected`, which is the same list as `example.expected.values()`? 
Do we need this block at all? can we just do 

```
            assert score[0].value == tuple(
                example.expected.values()), f"{example.predictions},{example.ground_truths}"
```
?

Reply:
We are actually accumulating values across all expected_values.
```
expected = [0, 0, 0, 0]
for expected_values in example.expected.values():
    for idx in range(4):
        expected[idx] += expected_values[idx]
assert ... # Each of the values are added to expected before this assert statement is reached.
```